spec.html

```diff
@@ -32027,79 +32027,65 @@ <h1>Time Values and Time Range</h1>
-        <p>The remainder is called the time within the day:</p>
-        <emu-eqn id="eqn-TimeWithinDay" aoid="TimeWithinDay">TimeWithinDay(_t_) = 𝔽(ℝ(_t_) modulo ℝ(msPerDay))</emu-eqn>
       </emu-clause>
 
       <emu-clause id="sec-year-number">
         <h1>Year Number</h1>
-        <p>ECMAScript uses a proleptic Gregorian calendar to map a day number to a year number and to determine the month and date within that year. In this calendar, leap years are precisely those which are (divisible by 4) and ((not divisible by 100) or (divisible by 400)). The number of days in year number _y_ is therefore defined by</p>
-        <emu-eqn id="eqn-DaysInYear" aoid="DaysInYear">
-          DaysInYear(_y_)
-            = *365*<sub>𝔽</sub> if (ℝ(_y_) modulo 4) &ne; 0
-            = *366*<sub>𝔽</sub> if (ℝ(_y_) modulo 4) = 0 and (ℝ(_y_) modulo 100) &ne; 0
-            = *365*<sub>𝔽</sub> if (ℝ(_y_) modulo 100) = 0 and (ℝ(_y_) modulo 400) &ne; 0
-            = *366*<sub>𝔽</sub> if (ℝ(_y_) modulo 400) = 0
-        </emu-eqn>
-        <p>All non-leap years have 365 days with the usual number of days per month and leap years have an extra day in February. The day number of the first day of year _y_ is given by:</p>
+        <p>ECMAScript uses a proleptic Gregorian calendar in which common years have 365 days and leap years (those that are divisible by 4 but not divisible by 100 and those that are divisible by 400) have 366 days by means of adding a 29th day to February. The day number of the first day of year _y_ changes by 365 days for every year of difference between _y_ and 1970 plus one day for each of those that is a leap year, and is defined as:</p>
         <emu-eqn id="eqn-DaysFromYear" aoid="DayFromYear">DayFromYear(_y_) = 𝔽(365 &times; (ℝ(_y_) - 1970) + floor((ℝ(_y_) - 1969) / 4) - floor((ℝ(_y_) - 1901) / 100) + floor((ℝ(_y_) - 1601) / 400))</emu-eqn>
         <p>The time value of the start of a year is:</p>
         <emu-eqn id="eqn-TimeFromYear" aoid="TimeFromYear">TimeFromYear(_y_) = msPerDay &times; DayFromYear(_y_)</emu-eqn>
         <p>A time value determines a year by:</p>
         <emu-eqn id="eqn-YearFromTime" aoid="YearFromTime">YearFromTime(_t_) = the largest integral Number _y_ (closest to +&infin;) such that TimeFromYear(_y_) &le; _t_</emu-eqn>
-        <p>The leap-year function is *1*<sub>𝔽</sub> for a time within a leap year and otherwise is *+0*<sub>𝔽</sub>:</p>
-        <emu-eqn id="eqn-InLeapYear" aoid="InLeapYear">
-          InLeapYear(_t_)
-            = *+0*<sub>𝔽</sub> if DaysInYear(YearFromTime(_t_)) = *365*<sub>𝔽</sub>
-            = *1*<sub>𝔽</sub> if DaysInYear(YearFromTime(_t_)) = *366*<sub>𝔽</sub>
-        </emu-eqn>
+        <p>The leap-year function is *1*<sub>𝔽</sub> for a leap year and otherwise is *+0*<sub>𝔽</sub>:</p>
+        <emu-eqn id="eqn-LeapDaysInYear" aoid="LeapDaysInYear" oldids="eqn-InLeapYear">LeapDaysInYear(_y_) = DayFromYear(_y_ + 1) - DayFromYear(_y_) - 365</emu-eqn>
```

Comment:
Does this run into problems in the year +275760, since DayFromYear is called on an out-of-range year? At first glance the DayFromYear definition seems to handle this OK.